packages/cli/src/active-executions.ts

```diff
@@ -125,68 +138,24 @@ export class ActiveExecutions {
 		execution?.responsePromise?.resolve(response);
 	}
 
-	getPostExecutePromiseCount(executionId: string): number {
-		return this.activeExecutions[executionId]?.postExecutePromises.length ?? 0;
-	}
-
-	/**
-	 * Remove an active execution
-	 */
-	remove(executionId: string, fullRunData?: IRun): void {
-		const execution = this.activeExecutions[executionId];
-		if (execution === undefined) {
-			return;
-		}
-
-		// Resolve all the waiting promises
-		for (const promise of execution.postExecutePromises) {
-			promise.resolve(fullRunData);
-		}
-
-		this.postExecuteCleanup(executionId);
-	}
-
-	/**
-	 * Forces an execution to stop
-	 */
+	/** Forces an execution to stop */
 	stopExecution(executionId: string): void {
-		const execution = this.activeExecutions[executionId];
-		if (execution === undefined) {
-			// There is no execution running with that id
-			return;
-		}
-
-		execution.workflowExecution!.cancel();
-
-		// Reject all the waiting promises
-		const reason = new ExecutionCancelledError(executionId);
-		for (const promise of execution.postExecutePromises) {
-			promise.reject(reason);
-		}
-
-		this.postExecuteCleanup(executionId);
+		const execution = this.getExecution(executionId);
+		execution.workflowExecution?.cancel();
+		execution.postExecutePromise.reject(new ExecutionCancelledError(executionId));
 	}
 
-	private postExecuteCleanup(executionId: string) {
-		const execution = this.activeExecutions[executionId];
-		if (execution === undefined) {
-			return;
-		}
-
-		// Remove from the list of active executions
-		delete this.activeExecutions[executionId];
-
-		this.concurrencyControl.release({ mode: execution.executionData.executionMode });
+	/** Mark an execution as completed */
+	finishExecution(executionId: string, fullRunData?: IRun) {
+		const execution = this.getExecution(executionId);
+		execution.postExecutePromise.resolve(fullRunData);
```

Comment:
```suggestion
	/** Resolve the post-execution promise in an execution. */
	settleExecution(executionId: string, fullRunData?: IRun) {		
    this.getExecution(executionId).postExecutePromise.resolve(fullRunData);
```

Reply:
Also maybe mention that resolving it with no data means failure.

Reply:
renamed `finishExecution` to `finalizeExecution`.

Regarding the comment explaining the no data scenario: I want to explore the option of rejecting the promise with other error types, like timeout, and any execution errors, and move away from resolving the promise on an error.